src/compiler/checker.ts

```diff
@@ -36776,31 +36777,33 @@ namespace ts {
                 return;
             }
 
-            if(node.flags & NodeFlags.ClassStaticBlockContext) {
-                grammarErrorOnFirstToken(node, Diagnostics.A_return_statement_cannot_be_used_inside_class_static_block);
+            const containingFunctionOrClassStaticBlock = getContainingFunctionOrClassStaticBlock(node);

```

Comment:
We could probably get away with a shorter name that has a similar meaning:
```suggestion
            const container = getContainingFunctionOrClassStaticBlock(node);
```